sdk/search/azure-search-documents/azure/search/documents/_service/_search_service_client.py

```diff
@@ -45,6 +45,14 @@ class SearchServiceClient(HeadersMixin):  # pylint: disable=too-many-public-meth
     def __init__(self, endpoint, credential, **kwargs):
         # type: (str, AzureKeyCredential, **Any) -> None
 
+        try:
+            if endpoint.lower().startswith('http') and not endpoint.lower().startswith('https'):
+                raise ValueError("Endpoint should be secure. Use https.")
```

Comment:
I would use a wording more consistent with Azure-core:
https://github.com/Azure/azure-sdk-for-python/blob/cd25de58e00d624511b5154e9fe8683826020bc0/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py#L55